### PR TITLE
feat: NoteEditor

### PR DESCRIPTION
**Pull request by vladyslav-baliuk**
Create `NoteEditor` component.

Screenshot 2024-05-15 at 15 47 17

### CODE REVIEW

**Comment by Letushev** [comment]
Finished Vlad's work that he started before the vacation.

File: src/components/note-editor/note-editor.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    description,
+    onDescriptionChange,
+    onDelete,
+    modifiedAt,
+    className,
+}: NoteEditorProps) => {
+    return (
+        <article className={classNames(styles.root, className)}>
+            <header className={styles.header}>
+                <Link to="/" className={styles.link}>
```
Comment (ynesteruk) [inline src/components/note-editor/note-editor.tsx:28]: Maybe LinkButton variant?

Comment (Letushev) [reply src/components/note-editor/note-editor.tsx:28]: It is just an `a`, no styles needed.

Comment (ynesteruk) [reply src/components/note-editor/note-editor.tsx:28]: ok

**Review by ynesteruk — APPROVED** [review]

**Pull request merged**